beets/dbcore/query.py

```diff
@@ -653,6 +653,35 @@ def col_clause(self):
         return clause, subvals
 
 
+class DurationQuery(NumericQuery):
+    """NumericQuery that allow human-friendly (M:SS) time interval formats.
+
+    Converts the range(s) to a float value, and delegates on NumericQuery.
+
+    Raises InvalidQueryError when the pattern does not represent an int, float
+    or M:SS time interval.
+    """
+    def _convert(self, s):
+        """Convert a M:SS or numeric string to a float.
+
+        Return None if `s` is empty.
+        Raise an InvalidQueryError if the string cannot be converted.
+        """
+        if not s:
+            return None
+        try:
+            # TODO: tidy up circular import
+            from beets.ui import raw_seconds_short
```

Comment:
Good point on the circular import—and we mostly try to make dbcore a separate component. Maybe the function should live in this module?